src/control/destination/background/destination_set.rs

```diff
@@ -31,10 +32,11 @@ use super::{ActiveQuery, DestinationServiceQuery, UpdateRx};
 
 /// Holds the state of a single resolution.
 pub(super) struct DestinationSet<T: HttpService<ResponseBody = RecvBody>> {
+    auth: Arc<DnsNameAndPort>,
```

Comment:
So, the existence of the destinationset can hold the reference that reconnects holds active, even when it's not in the destinations map?  I'd much prefer to avoid using shared pointers in this review if at all possible. There are subtleties like this that lead to complexity